storage/datastore.go

```diff
@@ -76,6 +76,17 @@ func (ds *DataStore) Read(txn Transaction, path ast.Ref) (interface{}, error) {
 	return ds.GetRef(path)
 }
 
+// Write modifies a document referred to by path.
+func (ds *DataStore) Write(txn Transaction, op PatchOp, path ast.Ref, value interface{}) error {
+	p, err := path.Underlying()
+	if err != nil {
+		return err
+	}
+	// TODO(tsandall): Patch() assumes that paths in writes are relative to
+	// "data" so drop the head here.
+	return ds.Patch(op, p[1:], value)
+}
```

Comment:
Currently, OPA will only write to the built-in store. When running as a daemon, this happens when users issue a PATCH request (on /data/...).